src/main/java/io/crate/jmx/http/HttpMetricHandler.java

```diff
@@ -67,15 +68,11 @@ public void handle(HttpExchange t) throws IOException {
 
         ByteArrayOutputStream response = this.response.get();
         response.reset();
-        OutputStreamWriter osw = new OutputStreamWriter(response, StandardCharsets.UTF_8);
-        TextFormat.write004(osw, registry.filteredMetricFamilySamples(parseQuery(query)));
-        osw.flush();
-        osw.close();
-        response.flush();
-        response.close();
+        try (OutputStreamWriter osw = new OutputStreamWriter(response, StandardCharsets.UTF_8)) {
+            TextFormat.write004(osw, registry.filteredMetricFamilySamples(parseQuery(query)));
+        }
 
         t.getResponseHeaders().set("Content-Type", TextFormat.CONTENT_TYPE_004);
-        t.getResponseHeaders().set("Content-Length", String.valueOf(response.size()));
```

Comment:
According to the docs: "If the response length parameter is zero, then chunked transfer encoding is used and an arbitrary amount of data may be sent."
But we set the response length if no using compression. We probably should set it to `0` when uncompressed as well? (or set the `Content-Length` header only in this case).

Reply:
> But we set the response length if not using compression. We probably should set it to 0 when uncompressed as well? (or set the Content-Length header only in this case).

That's fine, the important part is that we don't set content-length ourselves because `t.sendResponseHeaders(HttpURLConnection.HTTP_OK, response.size());` takes care of it.

For the compressed case we don't calculate the response size after compression so we use 0 for chunked transfer.
For the uncompressed case we know the content-length and can set it.

Reply:
Thanks for the clarification!